### PR TITLE
Add ItemWidth and ItemHeight properties to WrapPanel

### LINKED ISSUE #3003: Add ItemWidth and ItemHeight properties to WrapPanel

### PR DESCRIPTION
**Pull request by Kermalis**
## What does the pull request do?
Adds the `ItemWidth` and `ItemHeight` properties found in WPF's WrapPanel.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?

## Fixed issues
Closes #3003

### CODE REVIEW

**Comment by Kermalis** [comment]
The test is failing in the case of the second child. I'm not sure why given that WrapPanel is pretty much a line-for-line copy of WPF's at this point. The second child shows that its PreviousArrange is the correct size but the Bounds end up being something else somehow. In WPF the second child would be the size 20,15 as I try to set here with ItemWidth and ItemHeight. If Width and Height of the second child are "Auto"/double.Nan, then it works as intended. I think this is a bug somewhere else in Avalonia that someone else would have to check.

File: src/Avalonia.Controls/WrapPanel.cs
Comment (MarchingCube) [inline src/Avalonia.Controls/WrapPanel.cs]: Could you change all `constrint` to `constraint` in this file?

Comment (Kermalis) [reply src/Avalonia.Controls/WrapPanel.cs]: Resolved

**Comment by danwalmsley** [comment]
@Kermalis your unit tests appear to be failing.

**Comment by Kermalis** [comment]
Yeah, check my previous comment, I'm not sure why it's failing because the control is a copy of WPF's so the problem might be elsewhere

**Comment by Kermalis** [comment]
Since the test was failing due to some other issue, I will create an issue for it once this gets merged. The control works and should be merged

**Review by grokys — APPROVED** [review]
LGTM, if you could open the issue about the failing test, I'll put it on my list!

**Pull request merged**
